src/components/SelectionListSingle/BaseSelectionList.tsx

```diff
@@ -0,0 +1,341 @@
+            if (areSetsEqual(itemsToHighlight, newItemsToHighlight)) {
+                return;
+            }
+            const index = data.findIndex((option) => newItemsToHighlight.has(option.keyForList ?? ''));
+            scrollToIndex(index);
+            setItemsToHighlight(newItemsToHighlight);
+
+            if (itemFocusTimeoutRef.current) {
+                clearTimeout(itemFocusTimeoutRef.current);
+            }
+            itemFocusTimeoutRef.current = setTimeout(() => {
+                setItemsToHighlight(null);
+            }, ANIMATED_HIGHLIGHT_DURATION);
+        },
+        [data, itemsToHighlight, scrollToIndex],
+    );
+
+    useImperativeHandle(ref, () => ({scrollAndHighlightItem, scrollToIndex}), [scrollAndHighlightItem, scrollToIndex]);
+    return (
+        <View style={styles.flex1}>
+            {aboveListHeader()}
+            {data.length === 0 ? (
+                renderListEmptyContent()
+            ) : (
+                <>
+                    <SelectionListHeader
+                        dataDetails={dataDetails}
+                        aboveListHeaderMessage={aboveListHeaderMessage}
+                        customListHeader={customListHeader}
+                        canSelectMultiple={canSelectMultiple}
+                        onSelectAll={() => {
+                            onSelectAll?.();
+                            if (shouldShowTextInput && shouldPreventDefaultFocusOnSelectRow && innerTextInputRef.current) {
+                                innerTextInputRef.current.focus();
+                            }
+                        }}
+                        shouldPreventDefaultFocusOnSelectRow={shouldPreventDefaultFocusOnSelectRow}
+                    />
+                    <FlashList
+                        data={slicedData}
+                        renderItem={renderItem}
+                        ref={listRef}
+                        keyExtractor={(item, index) => `${item.keyForList}-${index}`}
+                        ListFooterComponent={
+                            listFooterContent ??
+                            (hasMoreDataToShow ? (
+                                <ShowMoreButton
+                                    containerStyle={[styles.mt2, styles.mb5]}
+                                    currentCount={PAGE_SIZE * currentPage}
+                                    totalCount={data.length}
+                                    onPress={incrementPage}
+                                />
+                            ) : null)
+                        }
+                        scrollEnabled={scrollEnabled}
+                        indicatorStyle="white"
+                        keyboardShouldPersistTaps="always"
+                        showsVerticalScrollIndicator={showScrollIndicator}
+                        onEndReached={onEndReached}
+                        onEndReachedThreshold={onEndReachedThreshold}
+                        style={listStyle as ViewStyle}
```

Comment:
I think FlashList made a mistake with typing, it should be `StyleProp<ViewStyle>` 😭 

[Made an issue upstream](https://github.com/Shopify/flash-list/issues/1889)